### PR TITLE
cpu: lpc2387: provide periph/* implementations, update to use newlib module.

### PR DESCRIPTION
**Pull request by kaspar030**
Our poor lpc2387-based boards are suffering heavy legacitis. This PR tries to cure them.

It includes GPIO, SPI and UART-implementations for lpc2387 and updates the architecture to our shared newlib module.

Heavily tested on msba2, unfortunately I don't have avsextreme or pttu boards.

This drops legacy cc110x and ltc4150 support.

~~**edit** waiting for #3620.~~

### CODE REVIEW

**Comment by OlegHahm** [comment]
Nice one! Will try to test ASAP with avsextrem and PTTU. Ping me if I forget to do so.

**Comment by kaspar030** [comment]
@OlegHahm *ping*

**Comment by OlegHahm** [comment]
How do I test GPIO?

**Comment by kaspar030** [comment]
I looked for an unused GPIO pin in the spec sheet and a short cable and connected the pin to +5v for the interrupt tests. for output, use a voltmeter between the pin and ground. (all using tests/periph_gpio).

File: boards/avsextrem/Makefile.features [hunk 1]
```diff
@@ -1,3 +1,4 @@
 FEATURES_PROVIDED += transceiver
+FEATURES_PROVIDED += periph_gpio
 FEATURES_PROVIDED += periph_rtc
```
Comment (haukepetersen) [inline boards/avsextrem/Makefile.features]: += periph_spi ?!

File: boards/msba2-common/Makefile.dep [hunk 2]
```diff
@@ -6,7 +6,3 @@ ifneq (,$(filter cc110x_legacy,$(USEMODULE)))
 	USEMODULE += cc110x_spi
 	USEMODULE += gpioint
```
Comment (haukepetersen) [inline boards/msba2-common/Makefile.dep]: gpioint is removed, right?

File: boards/avsextrem/Makefile.features (hunk 1 above)
Comment (kaspar030) [inline boards/avsextrem/Makefile.features]: fixed

File: boards/msba2-common/Makefile.dep (hunk 2 above)
Comment (kaspar030) [inline boards/msba2-common/Makefile.dep]: fixed

**Comment by OlegHahm** [comment]
GPIO seems to work.

**Comment by haukepetersen** [comment]
Looks good to me, everything I tried did also work (for msba2 and avsextrem). Although it's not 100% clean yet, I think it's a good step in the right direction -> so ACK when Travis is happy.

**Comment by haukepetersen** [comment]
Travis is failing to stuff related to the old network stack -> so let's wait until #3334 is merged an run Travis again!

**Comment by kaspar030** [comment]
- rebased
- squashed
- fixed doxygen warnings

**Comment by kaspar030** [comment]
(accidentaly removed radio_packet_length define, so this PR now depends on #3334)

**Comment by haukepetersen** [comment]
everything depends on #3334 :-)

**Comment by OlegHahm** [comment]
#3334 is history

**Comment by kaspar030** [comment]
I had to refactor the arm7 newlib support. Now only lpc2387-based devices are using the newlib module, the only other arm7 board, redbee-econotag, is using the cpu-specific newlibs.

@haukepetersen Does your ACK hold?

**Comment by kaspar030** [comment]
@haukepetersen *ping*

**Comment by haukepetersen** [comment]
Testing now

File: cpu/lpc2387/periph/spi.c [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+        cpol = 1;
+        break;
+    case SPI_CONF_SECOND_FALLING:/**< first data bit is transacted on the second falling SCK edge */
+        cpha = 1;
+        cpol = 1;
+        break;
+    }
+#endif
+
+    // Power
```
Comment (haukepetersen) [inline cpu/lpc2387/periph/spi.c]: c-style comments...

File: cpu/lpc2387/periph/spi.c [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+#endif
+#if SPI_1_EN
+    [SPI_1] = MUTEX_INIT,
+#endif
+#if SPI_2_EN
+    [SPI_2] = MUTEX_INIT
+#endif
+};
+
+int spi_init_master(spi_t dev, spi_conf_t conf, spi_speed_t speed)
```
Comment (haukepetersen) [inline cpu/lpc2387/periph/spi.c]: would be cool to put a TODO in here somewhere, that you don't set clock polarity and phase at the moment

File: cpu/lpc2387/periph/spi.c [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+    char tmp = (char)SSP0DR;
+
+    if (in != NULL) {
+        *in = tmp;
+    }
+
+    return 1;
+}
+
+int spi_transfer_bytes(spi_t dev, char *out, char *in, unsigned int length)
```
Comment (haukepetersen) [inline cpu/lpc2387/periph/spi.c]: this implementation is broken: when trying to use it full-duplex, it loses data...

**Comment by haukepetersen** [comment]
Did some tests for the msba2 and the avsextrem, everything I tried worked. I see some minor things I would like to see fixed:
- the `spi_transfer_bytes()` function is broken
- partly doxygen is missing (e.g. `periph_cpu.h`), I am wondering why Travis passed...
- please convert comments to c-style comments

ACK when these issues are addressed (and Travis is green again).

**Comment by haukepetersen** [comment]
oh, and some squashing wouldn't hurt...

**Comment by OlegHahm** [comment]
> partly doxygen is missing (e.g. periph_cpu.h), I am wondering why Travis passed...

Interesting question.

**Comment by kaspar030** [comment]
- moved newlib changes from arm7_common to just lpc2387

@haukepetersen does your ACK hold? travis is happy...

**Comment by kaspar030** [comment]
@haukepetersen Don't go to holidays without merging this! ;)

**Comment by haukepetersen** [comment]
Looking good -> ACK and go.

**Comment by OlegHahm** [comment]
:raised_hands:

**Comment by kaspar030** [comment]
nice. Thanks for reviewing!

**Pull request merged**